src/download.js

```diff
@@ -1,35 +1,79 @@
 /* eslint-disable no-unused-vars */
 
+import axios from "axios";
+
 import { SkynetClient } from "./client.js";
-import { makeUrlWithSkylink, defaultOptions } from "./utils.js";
+import { addUrlQuery, defaultOptions, makeUrl, parseSkylink, trimUriPrefix, uriHandshakePrefix } from "./utils.js";
 
 const defaultDownloadOptions = {
   ...defaultOptions("/"),
 };
+const defaultDownloadHnsOptions = {
+  ...defaultOptions("/hns"),
+};
 
+/**
+ * Initiates a download of the content at the skylink within the browser.
+ * @param {string} skylink - 46 character skylink.
+ * @param {Object} [customOptions={}] - Additional settings that can optionally be set.
+ * @param {string} [customOptions.endpointPath="/"] - The relative URL path of the portal endpoint to contact.
+ */
 SkynetClient.prototype.download = function (skylink, customOptions = {}) {
   const opts = { ...defaultDownloadOptions, ...customOptions, download: true };
   const url = this.getDownloadUrl(skylink, opts);
 
   window.open(url, "_blank");
 };
 
+SkynetClient.prototype.downloadHns = async function (hns, customOptions = {}) {
+  const opts = { ...defaultDownloadHnsOptions, ...customOptions };
+};
+
+SkynetClient.prototype.downloadHnsres = async function (hnsres, customOptions = {}) {
```

Comment:
```suggestion
SkynetClient.prototype.resolveHns = async function (domainName, customOptions = {}) {
```

Reply:
Hmm, it doesn't just resolve, it downloads too. Most accurate would be `resolveAndDownloadHns` but that's pretty long.